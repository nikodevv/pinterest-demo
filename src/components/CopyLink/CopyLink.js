import React from 'react';
import CopyText from "react-copy-text";
import {useDispatch} from "react-redux";
import {modalActionCreators} from "../../actions";

const CopyLink = (props) => {
  const [textToCopy, setTextToCopy] = React.useState(null);
  const dispatch = useDispatch();
  return <div className="screenOverlay container-row center"
              onClick={(e) => {
                if (e.currentTarget === e.target){
                  dispatch(modalActionCreators.toggleShortLinkModal());
                }
    }}>
    <div className="container-column center modalContainer">
      <div className="center container-row">
        <CopyText text={textToCopy} />
        {props.link}
      </div>
      <div className="center container-row">
        <button onClick={()=>setTextToCopy(props.link)}>Copy</button>
      </div>
      {textToCopy !== null && <div className="center container-row">
        <p>Link copied to clipboard!</p>
      </div>}
    </div>
  </div>;
};

export default CopyLink;
