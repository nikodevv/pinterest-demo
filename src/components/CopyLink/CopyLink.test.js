import React from 'react';
import {configure, shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import * as redux from "react-redux";
import Adapter from "enzyme-adapter-react-16/build";
import CopyLink from './CopyLink';

configure({adapter: new Adapter()});
const mockStore=configureMockStore()
describe('<CopyLink />', () => {
  test('renders', () => {
    const wrapper = shallow(
      <redux.Provider store={mockStore()}><CopyLink /></redux.Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
