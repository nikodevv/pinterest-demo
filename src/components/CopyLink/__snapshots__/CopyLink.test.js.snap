// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyLink /> renders 1`] = `ShallowWrapper {}`;
